python/ray/train/v2/api/data_parallel_trainer.py

```diff
@@ -86,6 +87,11 @@ def __init__(
         self.datasets = datasets or {}
         self.data_config = dataset_config or DataConfig()
 
+        self.running_in_local_mode = self.scaling_config.num_workers == 0
+        self.local_controller = None
+        if self.running_in_local_mode:
+            self.local_controller = self._get_local_controller()
```

Comment:
Why is this in `__init__`? Can we just move this initialization logic into `_initialize_and_run_local_controller`?